app/models/user.rb

```diff
@@ -48,6 +48,7 @@ module Levels
     _no_feedback
     disable_user_dmails
     enable_compact_uploader
+    use_new_search_syntax
```

Comment:
You can't insert new preferences anywhere, they're positional. It *has* to be at the end, else the preferences after it will be shifted around

Reply:
Fixed 9eff693ec34e2c879b4b2b78e82781af49330f99

Reply:
> I was unaware the order was important. I'll move it.

Ideally they wouldn't be, but instead of explicitly defining the values, we use the index value. The flag value is `1 << index`. 
Ex:
`1 << 0` = 1
`1 << 1` = 2
`1 << 2` = 4

These combine into a final number to say which are active, ex `3` (0 & 1), `5` (0 & 2), etc

Inserting a flag in the middle changes the index of later flags, so their value changes, making previous preferences now incorrect.
Ideally we'd explicitly define the value for each preference so the order doesn't matter, and so we can actually remove unused flags. The position being important is the reason those unused flags are still in there

Reply:
Got it. I'll keep that in mind for future PRs, though I doubt I'd be doing anything else preference related. I didn't even know it was a mask until you linked the `HasBitFlags` class with the

```
 define_method("#{attribute}?") do 
   send(field) & bit_flag > 0 
 end 
```

Part. I can't say I did an extremely deep dive to understand the code, which is why you saw a bit of copy pasting existing things and changing the content, like with the actual preference in the settings using the "raw" method, or the attributes, or the attribute accessors.